types/redux-first-router/index.d.ts

```diff
@@ -152,9 +183,15 @@ export interface Options {
     onAfterChange?<S>(dispatch: Dispatch<S>, getState: StateGetter): void;
     onBackNext?<S>(dispatch: Dispatch<S>, getState: StateGetter): void;
     restoreScroll?(history: History): ScrollBehavior;
+    initialDispatch?: boolean;
+    querySerializer?: QuerySerializer;
     navigators?: NavigatorsConfig;
 }
 
+// Params and Payload differ from Flow definition, as they can be arbitrarily shaped
+export type Params = any;
+export type Payload = any;
+
```

Comment:
thanks @faceyspacey - good to know, and happy to rework the typings after the big push, I'd like to get at least some notion of updated typings in so other folks don't have to sprinkle `any` typecasts all over 😄 

@Valbrand - according to [this](https://blog.mariusschulz.com/2017/02/24/typescript-2-2-the-object-type) link, the `object` type allows for numbers and strings too; I tried the generic `{ [key: string]: any }` approach but it doesn't jive with explicit non-index property names:

```
Types of property 'payload' are incompatible.
    Type 'Payload' is not comparable to type '{ id: string; }'.
      Property 'id' is missing in type 'Payload'.
```

For full context, here is the definition of `Payload`:

```
export type Payload = {
    [key: string]: any;
}
```

And here is an example of where I'm using it:

```
export interface RoutingAction {
    type: string;
    payload?: Payload;
}

export interface SomeAction extends RoutingAction {
    payload?: {
        id: string;
    };
}
```

I'm no TypeScript expert, so if I'm totally off base please enlighten me, but as far as I can tell, if I go with the `[key: string]` index type, upside is I can access anything inside the payload; downside is I can't actually make any sort of type assertions or extensions on the payload (all I know is I've got some string keys; no clue whether `id` or `slug` or `page` or anything is there or not). This breaks from how RFR works, because when I go to a specific route, I know the shape of the payload to a much higher degree of specificity than just an arbitrary key-value mapping, but I can't make any assertions because the abstract key-value typing is incompatible with the specific typing.

Given this, I propose to keep `Params` and `Payload` as `any`, pending the big changes coming to RFR and in lieu of the real solution that I'm convinced is out there but beyond my current knowledge. happy to discuss further!